stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/ChatDomainImpl.kt

```diff
@@ -337,7 +334,12 @@ internal class ChatDomainImpl internal constructor(
                 disconnect()
             }
         }
-        storeBgSyncDataWhenUserConnects()
+
+        if (backgroundSyncEnabled) {
+            client.setPushNotificationReceivedListener { channelType, channelId ->
+                offlineSyncFirebaseMessagingHandler.syncMessages(appContext, "$channelType:$channelId")
```

Comment:
What about cancelling pending jobs when user disconnects (just in case, should not happen very often)

```
    override suspend fun disconnect() {
        storeSyncState()
        job.cancelChildren()
        stopListening()
        stopClean()
        clearState()
        
        // cancel queued WorkManager job
    }
```

Reply:
Good catch! We should also cancel the worker and clear `PushNotificationsConfigStore` inside `ChatClient`